bin/locale-generator/locale-generator.php

```diff
@@ -0,0 +1,146 @@
+	unset( $locales->locales[ $key ]->cldr_nplurals );
+	$locales->locales[ $key ]->cldr_plural_expressions = array(
+		'zero'  => '',
+		'one'   => '',
+		'two'   => '',
+		'few'   => '',
+		'many'  => '',
+		'other' => '',
+	);
+
+	// Check the iso codes against the CLDR data in accending order.
+	if ( array_key_exists( $locales->locales[ $key ]->lang_code_iso_639_1, $cldr_locales ) ) {
+		$locales->locales[ $key ]->cldr_code = $locales->locales[ $key ]->lang_code_iso_639_1;
+	} elseif ( array_key_exists( $locales->locales[ $key ]->lang_code_iso_639_2, $cldr_locales ) ) {
+		$locales->locales[ $key ]->cldr_code = $locales->locales[ $key ]->lang_code_iso_639_2;
+	} elseif ( array_key_exists( $locales->locales[ $key ]->lang_code_iso_639_3, $cldr_locales ) ) {
+		$locales->locales[ $key ]->cldr_code = $locales->locales[ $key ]->lang_code_iso_639_3;
+	}
+
+	// If we found a matching CLDR code, add the plurals data to the GP locales.
+	if ( $locales->locales[ $key ]->cldr_code ) {
+		// Set the number of CLDR plurals.
+		$locales->locales[ $key ]->cldr_nplurals = count( $cldr_locales[ $locales->locales[ $key ]->cldr_code ] );
+
+		// Loop through the CLDR plural rules and set them according to their type.
+		foreach ( $cldr_locales[ $locales->locales[ $key ]->cldr_code ] as $type => $rule ) {
+			switch ( $type ) {
+				case 'pluralRule-count-zero':
+					$locales->locales[ $key ]->cldr_plural_expressions['zero'] = $rule;
+
+					break;
+				case 'pluralRule-count-one':
+					$locales->locales[ $key ]->cldr_plural_expressions['one'] = $rule;
+
+					break;
+				case 'pluralRule-count-two':
+					$locales->locales[ $key ]->cldr_plural_expressions['two'] = $rule;
+
+					break;
+				case 'pluralRule-count-few':
+					$locales->locales[ $key ]->cldr_plural_expressions['few'] = $rule;
+
+					break;
+				case 'pluralRule-count-many':
+					$locales->locales[ $key ]->cldr_plural_expressions['many'] = $rule;
+
+					break;
+				case 'pluralRule-count-other':
+					$locales->locales[ $key ]->cldr_plural_expressions['other'] = $rule;
+
+					break;
+			}
+		}
+	}
+}
+
+// Use the new GP Locales object to create output that can be added in to the locales.php file.
+generate_output( $locales );
+/**
+ * Function to output the locales data in PHP source format.
+ *
+ * @param GP_Locales $locales The GlotPress Locales object.
+ */
+function generate_output( $locales ) {
+	// Get all the GP_Locale variables we're going to output.
+	$vars = get_object_vars( $locales->locales['en'] );
+
+	// Loop through all of the locales.
+	foreach ( $locales->locales as $locale ) {
+		// Create the variable name we'll use to define the locale based on the slug with any dashes replaced with underscores.
+		$var_name = str_replace( '-', '_', $locale->slug );
+		$root_var_name = str_replace( '-', '_', $locale->variant_root );
+
+		// Output the first line to 'create' the GP_Locale object for this locale.
+		echo "\t\t\${$var_name} = new GP_Locale();\n"; // WPCS: XSS ok.
```

Comment:
Remove all `// WPCS: XSS ok.` comments and add the bin directory as a exclude-pattern to phpcs.xml.

Reply:
:+1: